src/datasets/utils/py_utils.py

```diff
@@ -200,18 +200,18 @@ def _single_map_nested(args):
         logging.set_verbosity_warning()
     # Print at least one thing to fix tqdm in notebooks in multiprocessing
     # see https://github.com/tqdm/tqdm/issues/485#issuecomment-473338308
-    if rank is not None and not disable_tqdm and "notebook" in tqdm.__name__:
+    if rank is not None and not disable_tqdm and any("notebook" in tqdm_cls.__name__ for tqdm_cls in tqdm.__mro__):
         print(" ", end="", flush=True)
 
     # Loop over single examples or batches and write to buffer/file if examples are to be updated
     pbar_iterable = data_struct.items() if isinstance(data_struct, dict) else data_struct
-    pbar_desc = "#" + str(rank) if rank is not None else None
+    pbar_desc = (desc or "") + " #" + str(rank) if rank is not None else desc
```

Comment:
this adds a space before the # if desc is None, but it's fine I think

Reply:
This line comes from `Dataset._map_single` and I just copy-pasted it here, so I'll also remove the space there.